ScreenControl_Unity/Assets/ScreenControl/Client/Scripts/Plugin System/InputActionManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Ultraleap.ScreenControl.Client
+{
+    // Class: InputActionManager
+    // The manager for all <ClientInputActions> to be handled and distributed. This runs the
+    // received data through any referenced <InputActionPlugins> and finaly distributes the data
+    // via the  <TransmitInputAction> event which should be listened to by any class hoping to make
+    // use of incoming <ClientInputActions>.
+    [DefaultExecutionOrder(-1)]
+    public class InputActionManager : MonoBehaviour
+    {
+        // Delegate: ClientInputActionEvent
+        // An Action to distribute a <ClientInputAction> via the <TransmitInputAction> event listener.
+        public delegate void ClientInputActionEvent(ClientInputAction _inputData);
+
+        // Variable: TransmitInputAction
+        // An event for transmitting <ClientInputActions> that have been modified via the active
+        // <plugins>
+        public static event ClientInputActionEvent TransmitInputAction;
+
+        // Variable: TransmitRawInputAction
+        // An event for transmitting <ClientInputActions> that have NOT been modified via any
+        // <plugins>
+        public static event ClientInputActionEvent TransmitRawInputAction;
+
+        public static InputActionManager Instance;
+
+        // Variable: plugins
+        // A pre-defined plugin array of <ToggleablePlugins> that modify incoming <ClientInputActions>
+        // based on custom rules.
+        [Tooltip("These plugins modify InputActions and are performed in order.")]
+        [SerializeField] ToggleablePlugin[] plugins;
```

Comment:
Is this array re-arrangable in the inspector UI? Not huge if not, just would be nicer UX.

Reply:
It will be in 2020+ (or 2021 not sure, Unity added rearranging Inspector arrays recently)